packages/sdk-js/src/plugins/devtools.ts

```diff
@@ -0,0 +1,229 @@
+  }
+
+  return (gb: GrowthBook | UserScopedGrowthBook | GrowthBookClient) => {
+    let payload = extractGbDebugPayload({ searchParams, requestCookies });
+
+    if (!payload && request) {
+      payload = extractGbDebugPayload({
+        searchParams: request.nextUrl.searchParams,
+        requestCookies: request.cookies,
+      });
+    }
+
+    let state: DevtoolsState = {};
+    if (payload) {
+      try {
+        state = JSON.parse(payload);
+      } catch (e) {
+        console.error("cannot parse devtools payload", e);
+      }
+    }
+
+    devtoolsPlugin(state)(gb);
+  };
+}
+
+/**
+ * Intended to be used with cookieParser() middleware from npm: 'cookie-parser'.
+ */
+export function devtoolsExpressPlugin({
+  request,
+}: {
+  request?: ExpressRequestCompat;
+}): Plugin {
+  return (gb: GrowthBook | UserScopedGrowthBook | GrowthBookClient) => {
+    let payload =
+      typeof request?.query?.["_gbdebug"] === "string"
+        ? request.query["_gbdebug"]
+        : undefined;
+    if (!payload) {
+      payload =
+        typeof request?.cookies?.["_gbdebug"] === "string"
+          ? request.cookies["_gbdebug"]
+          : undefined;
+    }
+
+    let state: DevtoolsState = {};
+    if (payload) {
+      try {
+        state = JSON.parse(payload);
+      } catch (e) {
+        console.error("cannot parse devtools payload", e);
+      }
+    }
+
+    devtoolsPlugin(state)(gb);
+  };
+}
+
+type LogUnionWithSource = LogUnion & { source?: string };
+type SdkInfo = {
+  apiHost: string;
+  clientKey: string;
+  version?: string;
+  payload?: FeatureApiResponse;
+  attributes?: Attributes;
+};
+type LogEvent = {
+  logs: LogUnionWithSource[];
+  sdkInfo?: SdkInfo;
+};
+/**
+ * Helper method to get log events for DevTools
+ * @param gb - GrowthBook instance. DevMode must be enabled to view log events.
+ * @param {string} [source] - Label these events for ease of reading in DevTools
+ * @example
+ * A React logger component (implement yourself):
+ ```
+  const event = getDebugEvent({ gb, "nextjs" });
+  return (
+    <script dangerouslySetInnerHTML={{
+      __html: \`(window._gbdebugEvents = (window._gbdebugEvents || [])).push(${JSON.stringify(event)});\`
+    }} />
+  );
+ ```
+ */
+export function getDebugEvent(
+  gb: GrowthBook | UserScopedGrowthBook,
+  source?: string
+): LogEvent | null {
+  if (!("logs" in gb)) return null;
+  if (gb instanceof GrowthBook) {
+    // GrowthBook SDK
+    const [apiHost, clientKey] = gb.getApiInfo();
+    return {
+      logs: gb.logs.map((log) => ({ ...log, source, clientKey })),
+      sdkInfo: {
+        apiHost,
+        clientKey,
+        version: gb.version,
+        payload: gb.getDecryptedPayload?.() || {
+          features: gb.getFeatures?.(),
+          experiments: gb.getExperiments?.(),
+        },
+        attributes: gb.getAttributes(),
+      },
+    };
+  } else if (gb instanceof UserScopedGrowthBook) {
+    // UserScopedGrowthBook SDK
+    // @ts-expect-error private access
```

Comment:
I don't love this, but not sure if worth addressing in the SDK. Should we make public getters?

Reply:
I don't think this will work.  During bundling, all variable names that start with underscore are mangled.  A public getter for both the parent GrowthBookClient instance and the `UserContext` makes sense.